sdk/core/core-tracing/src/createSpan.ts

```diff
@@ -48,6 +49,31 @@ export function isTracingDisabled(): boolean {
   return Boolean(azureTracingDisabledValue);
 }
 
+function pluckOptions<
+  T extends { tracingOptions?: OperationTracingOptions } & { spanOptions?: SpanOptions }
+>(operationOptions: T & { tracingOptions?: OperationTracingOptions }) {
+  const { tracingOptions, spanOptions, ...otherOptions } = operationOptions || {};
+
+  return {
+    tracingOptions,
+    spanOptions,
+    otherOptions
+  };
+}
+
+function startSpan(spanName: string, spanOptions?: SpanOptions, context?: Context) {
+  const mergedSpanOptions: SpanOptions = {
+    kind: SpanKind.INTERNAL,
+    ...spanOptions
+  };
+
+  if (isTracingDisabled()) {
```

Comment:
seems like this check can go to the top of the function, why create `mergedSpanOptions` otherwise?